### PR TITLE
quick proof of concept for server mod sending.

### PR DESCRIPTION
**Pull request by red-001**
proof of concept that shows how minetest#5965 will help with implementing server mod sending.

### CODE REVIEW

File: src/client.cpp [hunk 1]
```diff
@@ -217,17 +218,16 @@ void Client::loadModSubfolder(const std::string &mod_name, const std::string &mo
...
 	// If modding is not enabled, don't load mods, just builtin
 	if (!m_modding_enabled) {
 		return;
 	}
 
 	// Load and run "mod" scripts
-	for (std::vector<ModSpec>::const_iterator it = m_mods.begin();
-			it != m_mods.end(); ++it) {
-		const ModSpec &mod = *it;
+	for (ModSpec mod : m_mods) {
```
Comment (nerzhul) [inline src/client.cpp]: const ref

Comment (red-001) [reply src/client.cpp]: it's only a proof of concept it's not indented to merged which out major improvements.

File: builtin/mainmenu/dlg_settings_advanced.lua [hunk 2]
```diff
@@ -264,7 +264,7 @@ end
 -- read_all: whether to ignore certain setting types for GUI or not
 -- parse_mods: whether to parse settingtypes.txt in mods and games
 local function parse_config_file(read_all, parse_mods)
-	local builtin_path = core.get_builtin_path() .. DIR_DELIM .. FILENAME
+	local builtin_path = core.get_builtin_path() .. FILENAME
```
Comment (red-001) [inline builtin/mainmenu/dlg_settings_advanced.lua]: test

Comment (red-001) [inline builtin/mainmenu/dlg_settings_advanced.lua]: test

Comment (red-001) [inline builtin/mainmenu/dlg_settings_advanced.lua:267]: test 3

File: builtin/mainmenu/init.lua [hunk 3]
```diff
@@ -27,12 +27,12 @@ local basepath = core.get_builtin_path()
 defaulttexturedir = core.get_texturepath_share() .. DIR_DELIM .. "base" ..
```
Comment (red-001) [inline builtin/mainmenu/init.lua:27]: test

**Pull request closed without merging**